SimulationAgent/SimulationManager.cs

```diff
@@ -93,12 +97,14 @@ public class SimulationManager : ISimulationManager
             IFactory factory,
             IClusteringConfig clusteringConfig,
             ILogger logger,
-            IInstance instance)
+            IInstance instance,
+            ISimulationStatistics simulationStatistics)
```

Comment:
Does this need to be passed in? Is there a reason that this can't be a singleton for the entire process, or at least just a field of this class?

Reply:
This does not need to be a singleton, it is not maintaining any state and is used to read/write data from storage.